custom/icds/templates/partials/tableau_js.html

```diff
@@ -0,0 +1,213 @@
+<script>
+    var viz = {};
+    var workbook = {};
+    function initializeViz() {
+        var placeholderDiv = document.getElementById("tableauPlaceholder");
+        var url = "https://icds.commcarehq.org/#/views/{{ report_workbook }}/{{ report_worksheet }}";
+        var options = {
+            width: placeholderDiv.offsetWidth,
+            height: placeholderDiv.offsetHeight,
+            hideTabs: true,
+            hideToolbar: true,
+            onFirstInteractive: function () {
+                setUpNav(viz)
+            }
+        };
+        viz = new tableau.Viz(placeholderDiv, url, options);
+        // I am envisioning the history object possibly containing things like the base params for filtering
+        var historyObject = {
+            'location': '{{ report_worksheet }}'
+        };
+        history.pushState(historyObject, '', '{{ report_worksheet }}')
+    }
+
+    function setUpNav(viz) {
+        workbook = viz.getWorkbook();
+        var sheets = workbook.getPublishedSheetsInfo();
+        _.each(sheets, function(sheet) {
+            // TODO: This is a bit hacky, $0 was the only item in the sheet object though (maybe has to do with dashboards vs sheets?)
+            addNavigationLink(sheet.$0.name, sheet.$0.isActive)
+        });
+
+        // Apply inital params for user level filtering
+        var params = {
+            'user_view_by': '{{ user_view_by }}',
+            'user_state': '{{ user_state|default:'' }}',
+            'user_district': '{{ user_district|default:'' }}',
+            'user_block': '{{ user_block|default:'' }}',
+            'user_supervisor': '{{ user_supervisor|default:''}}',
+            'user_awc': '{{ user_awc|default:'' }}'
+        };
+        applyParams(workbook, params);
+
+        $(".nav-link").click(function (){
+            var link = $(this);
+            workbook.activateSheetAsync(link[0].textContent);
+        });
+
+        viz.addEventListener(tableau.TableauEventName.MARKS_SELECTION, onMarksSelection);
+    }
+
+    function addNavigationLink(sheetName, isActive) {
+        var btnColor = isActive  ? 'success' : 'default';
+        // var html = "<div class='row'><button class='btn btn-block btn-" + btnColor + "'>" + sheetName + "</btn></div>";
+        var html = "<li><a href='#' class='nav-link'>" + sheetName + "</a></li>";
+        $(".dropdown-menu").append(html);
+    }
+
+    function onMarksSelection(marksEvent) {
+        return marksEvent.getMarksAsync().then(activateDrilldown);
+    }
+
+    function activateDrilldown(marks) {
+        clearDrilldown();
+        // marks is an array of what the user has clicked
+        var html = [],
+            sheetName = '',
+            filters = {},
+            params = {},
+            filter_key,
+            filter_value;
+        for (var markIndex = 0; markIndex < marks.length; markIndex++) {
```

Comment:
You can also use pure javascript map or forEach: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach